comp/core/autodiscovery/listeners/snmp.go

```diff
@@ -181,20 +223,21 @@ func (l *SNMPListener) checkDevices() {
 		startingIP := ipAddr.Mask(ipNet.Mask)
 
 		configHash := config.Digest(config.Network)
-		cacheKey := fmt.Sprintf("snmp:%s", configHash)
+		cacheKey := fmt.Sprintf("%s:%s", cacheKeyPrefix, configHash)
 		adIdentifier := config.ADIdentifier
 		if adIdentifier == "" {
 			adIdentifier = "snmp"
 		}
 
 		subnet := snmpSubnet{
-			adIdentifier:   adIdentifier,
-			config:         config,
-			startingIP:     startingIP,
-			network:        *ipNet,
-			cacheKey:       cacheKey,
-			devices:        map[string]string{},
-			deviceFailures: map[string]int{},
+			adIdentifier:          adIdentifier,
+			config:                config,
+			startingIP:            startingIP,
+			network:               *ipNet,
+			cacheKey:              cacheKey,
+			devices:               map[string]string{},
+			deviceFailures:        map[string]int{},
+			devicesScannedCounter: *atomic.NewUint32(0),
```

Comment:
nit: this is not required, devicesScannedCounter is of type atomic.Uint32 so it's always initialized at 0